main.tf

```diff
@@ -72,6 +72,14 @@ resource "aws_cloudfront_distribution" "default" {
       origin_read_timeout      = var.origin_read_timeout
     }
 
+    dynamic "origin_shield" {
+      for_each = var.origin_shield_enabled ? ["true"] : []
```

Comment:
What if you enabled it and then disabled it? it looks like it would remove the block if it was disabled, but would it actually disable it if the block was removed? or would enabled have to be explicitly set to origin_shield_enabled without the for_each?

Reply:
The API isn't very consistent. I've tried two ways of doing this and enabling/disabling, both seem to require a third step to apply cleanly.

1. Use a dynamic `origin_shield` block

The Origin shield is not disabled when the block is removed. Applying again will cause the same changes to be planned.

2. Using a persistent `origin_shield` block

When setting `enabled` to `false` it works. Applying again will cause the plan to want to add in the `origin_shield` block again.

I'm putting in a change now to have an `origin_shield` variable that is an object to match the existing provider definition more closely.

Reply:
This is now a two-step process to remove, which is the same as the underlying resource.

1. Set `origin_shield.enabled` to `false.
2. Apply
3. Remove the `origin_shield` variable.
4. Apply